lib/internal/test_runner/test.js

```diff
@@ -176,22 +176,118 @@ function testMatchesPattern(test, patterns) {
 }
 
 class TestPlan {
-  constructor(count) {
-    validateUint32(count, 'count');
+  #waitIndefinitely = false;
+
+  constructor(count, options = kEmptyObject) {
+    this.#validateConstructorArgs(count, options);
     this.expected = count;
     this.actual = 0;
+    this.#setWaitOption(options);
+  }
+
+  #validateConstructorArgs(count, options) {
+    validateUint32(count, 'count');
+    validateObject(options, 'options');
+  }
+
+  #setWaitOption(options) {
```

Comment:
Can you inline this in the constructor as well. I think it would be more readable as an `if...else`. You should also destructure the options out so that we are only reading the value once. Otherwise, users can pass an object with bizarre getters that change the value of the option after we have validated it.

```js
const { wait } = options;
if (typeof wait === 'boolean') {
  //
} else if (typeof wait === 'number') {
  //
} else if (wait !== undefined) {
  // throw
}
```